p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -3484,11 +3484,13 @@ program, with decreasing priority levels.
 
 ####	Additional table properties
 
-Tables can have additional properties. The P4 spec does not mandate any additional properties, or prescribe their semantics. Various architectures can require attributes that are specific to these architectures.
+Tables can have additional properties. This specification does not mandate any additional properties, or prescribe their semantics. Various architectures can require properties that are specific to these architectures.
 
-For example, architectures where lookup-table resources are statically allocated may mandate a ```size``` table attribute, which can be used to indicate to the compiler back-end how many storage resources should be allocated.
+For example, architectures where lookup-table resources are statically allocated may mandate a ```size``` table property, which can be used to indicate to the compiler back-end how many storage resources should be allocated.
 
-A ```table``` declaration indicates the interfaces expected from a ```table```: keys and actions. However, the best way to implement a table is actually dependent on the nature of the entries that will populate the table at runtime (for example, tables could be dense or sparse, could be implemented as hash-tables, associative memories, tries, etc.) An implementation attribute could also be used to pass additional information to the compiler back-end. The value of this attribute could be an instance of an ```extern``` block chosen from a suitable library of components. For example, the core functionality of the P4~14~ table ```action_profile``` constructs could be implemented on architectures that support this feature using a construct such as the following:
+Additional properties are permitted to change the semantics of a table lookup. For example, an architecture could define a property ```default_action_undefined``` that, when given the value ```true```, causes the behavior of the data plane to be undefined for any packet that found no match when searching the table.
```

Comment:
Andy's text is consistent with what was discussed at the language design meeting yesterday. Can you suggest an appropriate rewording that would let the compiler do its job?